internal/util/openapi/openapi.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+	"sigs.k8s.io/kustomize/kyaml/openapi/kustomizationapi"
+)
+
+// ConfigureOpenAPI sets the openAPI schema in kyaml. It can either
+// fetch the schema from a cluster, read it from file, or just the
+// schema built into kyaml.
+func ConfigureOpenAPI(factory util.Factory, k8sSchemaSource, k8sSchemaPath string) error {
```

Comment:
How does this ensure that the schema is not being overridden in kyaml. For example, if I use 

`kpt pkg update --strategy resource-merge --k8s-schema-source cluster`,  how are we guaranteeing that the schema is not overridden by [initSchema()](https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/openapi/openapi.go#L328) in kyaml.